clients/lite-xl/tests/init_completion_resolve_test.lua

```
@@ -0,0 +1,880 @@
+  }
+  function server:get_language_id() return "perl" end
+  function server:exit() self.exited = (self.exited or 0) + 1 end
+  function server:get_completion_item_kind() return 1 end
+  function server:log(fmt) log_records[#log_records + 1] = tostring(fmt) end
+  local function record(kind, method, entry)
+    entry.method = method
+    entry.kind = kind
+    server.outbound[#server.outbound + 1] = entry
+  end
+  function server:push_notification(method, entry) record("notification", method, entry) end
+  function server:push_request(method, entry)
```

Luacheck (1.2.0)

[warning] unused argument 'self'

(W212)